src/org/wso2/tg/jenkins/executors/TestGridExecutor.groovy

```diff
@@ -40,6 +41,43 @@ def generateTesPlans(def product, def configYaml) {
     """
 }
 
+/**
+ * Select the schedule considering build cause. If the build has been triggered by a user return
+ * the schedule as manual and otherwise return time based period schedule.
+ *
+ * @param build current build
+ * @return schedule the schedule use for generate infrastructure combinations
+ */
+static def selectSchedule(build) {
+    def props = Properties.instance
+    // Check if the build was triggered by some jenkins user
+    def userId = build.rawBuild.getCause(Cause.UserIdCause.class).properties.userId
```

Comment:
Accessing the rawBuild requires to go beyond the Groovy sandbox in Jenkins. We usually have to add '@NonCPS' annotation to groovy methods that need to over-ride groovy sandbox. 

I wonder why it was not needed here!

Reply:
Tests are done with the groovy sandbox and the method used in here worked fine.